source/manual/content-security-policy.html.md

```diff
@@ -14,21 +14,74 @@ the browser.
+scripts and styles (in applicable applications) before launching it.
+
+[unsafe-inline]: https://content-security-policy.com/unsafe-inline/
+
 ## How the policy is set
 
-The specific policy that is sent with a request to the browser as an HTTP header is defined in the
-[`govuk_app_config` gem](https://github.com/alphagov/govuk_app_config/blob/master/lib/govuk_app_config/govuk_content_security_policy.rb)
-which is included in all frontend apps. This central definition means the entire site has a single policy,
-and changes can be rolled out more easily.
+GOV.UK has a base policy set in the [`govuk_app_config` gem][govuk_csp]. This policy is shared amongst GOV.UK
+applications and should contain directives that are either global or common to many applications.
 
-Each frontend app has an [initialiser](https://github.com/alphagov/government-frontend/blob/master/config/initializers/csp.rb)
-which invokes the CSP setting code in the gem.
+Each frontend app has an [initialiser](https://github.com/alphagov/government-frontend/blob/main/config/initializers/csp.rb)
+which invokes the CSP setting code in the gem. Each application can make customisations to the policy for
+application/route specific needs ([example in Smart Answers](https://github.com/alphagov/smart-answers/blob/1a2ff1d9f430afcc7435ac9775cc44de6b0a98f1/app/controllers/smart_answers_controller.rb#L8-L12)).
 
-## How violations are reported
+[govuk_csp]: https://github.com/alphagov/govuk_app_config/blob/main/lib/govuk_app_config/govuk_content_security_policy.rb
 
-In all production-like environments (production, staging, integration), CSP is running in report only mode while more
-information is gathered about which parts of the site currently rely on behaviour that would be blocked by an enforced
-CSP.
+## How violations are reported
 
-In other environments such as development and test, CSP is running in enforcement mode to allow errors to be captured
+In all production-like environments (production, staging, integration), CSP is running in report only mode. In other
+environments such as development and test, CSP is running in enforcement mode to allow errors to be captured
 at an early stage.
+
+We log reports to Amazon S3 bucket which can be queried with [Athena](https://aws.amazon.com/athena/). We store them
+for 30 days. Many of the reports we receive are from browser extensions we can't control so we
+[filter the most prolific of them][lambda] from our logs.
+
+As we receive high volumes of false positive alerts, it is likely we will remove the reporting functionality once
+a CSP is enforced.
+
+[lambda]: https://github.com/alphagov/govuk-aws/blob/main/terraform/lambda/CspReportsToFirehose/index.mjs
+
+### Querying violations
+
+Athena is available through the AWS control panel. To access, [log into AWS](/manual/get-started.html#sign-in-to-aws),
+navigate to [Athena](https://eu-west-1.console.aws.amazon.com/athena) and select the `csp_reports` database. The
```

Comment:
When you go to athena, there's a pre-query landing page where it took me a good 30 seconds to work out that I had to choose a query before I got to the point where I could choose "csp_reports" - possibly mention that?

(Also, on production this works fine, but on integration you can't run a query yet because "Before you run your first query, you need to set up a query result location in Amazon S3.")

Reply:
Ooh what do you get when you visit https://eu-west-1.console.aws.amazon.com/athena ? I get the query editor 

Yeah I'm trying to get the latter sorted in all environments with: https://github.com/alphagov/govuk-aws/pull/1668

Reply:
Ah, _now_ I get the query editor, but the first time I did it I got bounced to the Athena landing page (https://eu-west-1.console.aws.amazon.com/athena/home?region=eu-west-1#/landing-page). So it's obviously only a thing if you've never used Athena before (but I might argue most people suddenly reading this on 2L or wherever might not have used Athena before).

Reply:
Great thanks for sharing. I'll update.

Reply:
I've pushed [d5e5b55](https://github.com/alphagov/govuk-developer-docs/pull/3795/commits/d5e5b55366420c71079ee7686359a38daa42c816) which now uses an anchor to link directly to the query editor.